client/rest/src/main/java/org/opensearch/client/RequestLogger.java

```diff
@@ -67,11 +68,10 @@ private RequestLogger() {}
      * Logs a request that yielded a response
      */
     static void logResponse(Log logger, HttpUriRequest request, HttpHost host, ClassicHttpResponse httpResponse) {
-        if (logger.isDebugEnabled()) {
-            logger.debug(
-                "request [" + request.getMethod() + " " + host + getUri(request) + "] returned [" + new StatusLine(httpResponse) + "]"
-            );
-        }
+        logger.debug("request [{} {}{}] returned [{}]", () -> request.getMethod(), () -> host,
+        () -> getUri(request),
+        () -> new StatusLine(httpResponse));
+
```

Comment:
why not use 
```
logger.debug(() -> new ParameterisedMessage("Log blah {}"), blah); 
```

Reply:
Hi @Bukhtawar, Thanks for sharing inputs.

The lambda expression itself is responsible for creating a ParameterizedMessage instance and formatting the log message. So, in the statement `logger.debug(() -> new ParameterizedMessage("Log blah {}", blah));` you suggested, `ParameterizedMessage` constructor and message formatting will be executed when this statement is executed, regardless of the log level.

However, in `logger.debug("Log blah {}", () -> blah)`, the message template "Log blah {}" is provided as a constant string and the lambda expression `() -> blah` will only be executed if the debug log level is enabled. This allows for lazy computation, as `blah` is evaluated only when necessary. 

The lazy computation is what is desired, hence, I'd recommend the way I originally recommended. If there is some information which you think I have not accounted for, please feel free to share.